app/src/main/java/at/roteskreuz/stopcorona/model/repositories/ExposureNotificationRepository.kt

```diff
@@ -93,6 +94,26 @@ interface ExposureNotificationRepository {
      * in a blocking manner.
      */
     suspend fun getTemporaryExposureKeys(): List<TemporaryExposureKey>
+
+    /**
+     * Register with the Google Exposure Notifications framework in a blocking manner.
+     */
+    //TODO: for Dusan, see if we actually nee the non blocking function
+    suspend fun registerAppForExposureNotificationsNow()
+
+    /**
+     * had over Diagnosis Key files to the framework for
+     */
+    suspend fun processBatchDiagnosisKeys(archives: List<File>, token: String): String
+
+    /**
+     * use the [ExposureNotificationClient.getExposureSummary] to check if the batch is GREEN or
```

Comment:
You are describing implementation detail, which should in an implementation method. This is an interface which should hide implementation detail as much as possible.

Reply:
@falkorichter